### PR TITLE
Fixed errors in base data fitting pipeline

### PR DESCRIPTION
**Pull request by pamede**
With this I fixed errors in the base data fitting pipeline. For this I added a script that creates single pencil beam stf structs. I was only able to work with Octave as I do not have a MatLab version available at the moment, but the code should work in MatLab, too. In the baseDataFitting function there is now a switch to use different optimizers depending on whether one is using Octave or MatLab. 
I am not completely happy with the selection of whether to use the "BDL_matRad.txt" MCsquare base data file or fitting a new one in  matRad_calcParticleDoseMCsquare. For now I implemented it with a hardcoded switch. What could be a better alternative here?

Additionaly i fixed a dimension error in the resizeToCstGrid function that I noticed.

### CODE REVIEW

File: examples/matRad_example13_fitAnalyticalParticleBaseData.m
Comment (wahln) [inline examples/matRad_example13_fitAnalyticalParticleBaseData.m:104]: I don't really like this property, since apart from this specific use case, it has no meaning at all. My idea for a change would be to create a dummy machine file that does not contain the tabulated data, but all the metadata, as is done in line 144. I will comment further down.

File: examples/matRad_example13_fitAnalyticalParticleBaseData.m
Comment (wahln) [inline examples/matRad_example13_fitAnalyticalParticleBaseData.m:136]: Fitting with constRBE does not make sense. I would rather set quantityOpt pyhsicalDose & modelname none

File: examples/matRad_example13_fitAnalyticalParticleBaseData.m
Comment (wahln) [inline examples/matRad_example13_fitAnalyticalParticleBaseData.m:152]: This is good! I would complement it with reading the MCsquare base data text file as done from l 159 on, and writing all phase space information into there (without the tabluated IDD & sigmas etc.). Afterwards you save the dummy machine file to a mat file, which would then go through the normal loading process without issues or intrusive code changes.

File: examples/matRad_example13_fitAnalyticalParticleBaseData.m
Comment (wahln) [inline examples/matRad_example13_fitAnalyticalParticleBaseData.m:156]: If you use the dummy machine file as suggested, you don#t need the extra SAD argument.

File: examples/matRad_example13_fitAnalyticalParticleBaseData.m
Comment (wahln) [inline examples/matRad_example13_fitAnalyticalParticleBaseData.m:164]: as suggested above, write this into the temporary/dummy mat file

File: matRad_calcDoseInit.m
Comment (wahln) [inline matRad_calcDoseInit.m:11]: as suggested above, this should be removed

File: matRad_calcDoseInit.m
Comment (wahln) [inline matRad_calcDoseInit.m:168]: when you use the dummy file as suggested, you could just use this loading process as usual.

File: matRad_calcParticleDoseMCsquare.m
Comment (wahln) [inline matRad_calcParticleDoseMCsquare.m:187]: This here I don't really understand. Why is there a hardcoded 1 here?

Ah, I might understand now, that the changes regarding the emittance structure are not in here. We probably should put this on hold until the restructured emittance is merged into this.

File: matRad_generateStfSinglePencilBeam.m
Comment (wahln) [inline matRad_generateStfSinglePencilBeam.m:1]: see above comment regarding the SAD argument.

File: tools/matRad_fitBaseData.m
Comment (wahln) [inline tools/matRad_fitBaseData.m:107]: If the function definitions are the same in the Matlab & Octave branch, maybe you can save the handles first and then make the if-clause. Now they are implemented twice, right?

**Review by wahln — CHANGES_REQUESTED** [review]
I guess there's two major things here:
- Instead of explicitly stating that we do not want to run with a machine file, I would create a dummy machine file containing all the information (incl. phase space as in #470 ), save it, such that we do not need to catch this for this singular application at the loading process
- We probably should wait for #470 -> dev_MonteCarlo -> dev_varRBErobOpt before finalizing this, to have full access to the information during MC dose calculation als proposed there.

File: examples/matRad_example13_fitAnalyticalParticleBaseData.m
Comment (pamede) [reply examples/matRad_example13_fitAnalyticalParticleBaseData.m]: Yes, is deleted now.

File: examples/matRad_example13_fitAnalyticalParticleBaseData.m
Comment (pamede) [reply examples/matRad_example13_fitAnalyticalParticleBaseData.m]: Property was removed and replaced with the mentionen creation of a dummy machine file.

File: examples/matRad_example13_fitAnalyticalParticleBaseData.m
Comment (pamede) [reply examples/matRad_example13_fitAnalyticalParticleBaseData.m]: Done.

File: examples/matRad_example13_fitAnalyticalParticleBaseData.m
Comment (pamede) [reply examples/matRad_example13_fitAnalyticalParticleBaseData.m]: see above

File: examples/matRad_example13_fitAnalyticalParticleBaseData.m
Comment (pamede) [reply examples/matRad_example13_fitAnalyticalParticleBaseData.m]: Fixed this using the dummy file.

File: matRad_calcDoseInit.m
Comment (pamede) [reply matRad_calcDoseInit.m]: Done.

File: matRad_generateStfSinglePencilBeam.m
Comment (pamede) [reply matRad_generateStfSinglePencilBeam.m]: Done.

File: matRad_calcParticleDoseMCsquare.m
Comment (pamede) [reply matRad_calcParticleDoseMCsquare.m]: The problem here is that one needs to prevent the standard BDL file fitting process in a fitting run. This is why the switch is here. With the latest commit I turned it back to zero to ensure normal functionality in all other cases apart from the fitting use. In addition to that I added a comment in the fitting example, explaining the need to change the switch accordingly for the fitting process. Do you have an idea how to solve this problem in a more elegant way?

File: tools/matRad_fitBaseData.m
Comment (pamede) [reply tools/matRad_fitBaseData.m:107]: Here, the octave and matlab functions are not all precisely the same, as Octave does not allow to use the keyword "all" in the sum function. This is why I did not move the function declerations out of the if statement.

**Comment by wahln** [comment]
This PR has been merged into #536  and will be integrated with it, so I will close this now

**Pull request closed without merging**
